storm-client/src/jvm/org/apache/storm/coordination/BatchSubtopologyBuilder.java

```diff
@@ -446,31 +446,26 @@ public String getComponent() {
         }
         
         private void addDeclaration(InputDeclaration declaration) {
-            _component.declarations.add(declaration);
+            component.declarations.add(declaration);
         }
 
         @Override
         public BoltDeclarer addConfigurations(Map<String, Object> conf) {
-            _component.componentConfs.add(conf);
+            if (conf != null && !conf.isEmpty()) {
```

Comment:
Nit: I think putAll is a no-op when the conf is empty, so you probably don't need to check.